### ISSUE TITLE
Restructure the client-side API for running analyses

### ISSUE THREAD

**Issue by jcmorrow** [RecordReplay/devtools#6852]
See https://github.com/RecordReplay/devtools/issues/6591 for a lot more context here.

### Description of this task

This is about how do perform the commands that all analyses will have run (create, find, run, release) and how can we interact with that imperative process (for example, can we please get a promise API, rather than passing in `onPoints` or `onResults` callbacks?)

**Comment by jcmorrow** [comment]
@markerikson I think this is in-progress yeah? I'm gonna mark it as in-progress, feel free to change it if that's not accurate. And also it would be great to get an update and/or one of those heartbeat Discord threads that you are so good at! 💖

**Comment by markerikson** [comment]
For varying values of "in progress", technically yes :)

Actual status atm:

- We talked Wednesday
- Thursday I was digging through code, ended up over in `client/commands.js`, and figured that refactoring it to TS would be a good starting point to get my mind moving since I'd be dealing with actual code.  Put up that PR.
- Spent all of Friday fighting with CI/E2E and backing out changes to the PR until I stopped breaking the E2E tests.  Finally merged that.
- Today: meetings, annoyingly interspersed at the wrong intervals, and I've been completely unable to actually get going on real coding work.  So, today's basically a lost cause.

I currently have no meetings on the calendar tomorrow other than the daily standup, so my goal is to fully dig in on this.  I think that I have my head wrapped around the overall goals and problem space pretty well, and Brian's PR is excellent and provides a good basis to compare against.  But I really need to have uninterrupted time to get into the code and focus on this.
